### PR TITLE
Shared Input component

### LINKED ISSUE #549: [UI Components] Input Text Boxes
Create reusable free text input box components similar to the one in data entry page, but allow for it to be customizable within

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Shared input component

## Type of change
Type: Non-breaking refactor

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #549 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/Settings/AgencySettingsTeamManagement.tsx [hunk 1]
```diff
@@ -204,18 +206,38 @@ export const AgencySettingsTeamManagement = observer(() => {
...
+                      setInputsError({
+                        message: "Please enter a valid email.",
+                      });
+                    } else {
+                      setInputsError(undefined);
+                    }
+                  }}
+                  error={inputsError}
+                  textSize="small"
+                  // error={!!emailValue && !validateEmail(emailValue)}
```
Comment (mxosman) [inline publisher/src/components/Settings/AgencySettingsTeamManagement.tsx:240]: Safe to delete?

**Review by mxosman — COMMENTED** [review]
Ivan - amazing as always. I had one question about the `NotReportIcon` but otherwise - this looks fantastic to me. Thank you so much for the well thought-out  & clean implementation!

File: common/components/Input/NotReportedIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        <Styled.NotReportedIconTooltipHoverArea size={size}>
+          <Styled.NotReportedIconTooltip>
+            This has been disabled by an admin because the data is unavailable.
+            If you have the data for this, consider changing the configuration
+            in the{" "}
+            <Styled.TooltipLink onClick={notReportingTooltipLink}>
+              Settings
+            </Styled.TooltipLink>
+            .
+          </Styled.NotReportedIconTooltip>
```
[the remark this answers is not in the document]
Comment (ivan-kishko) [reply common/components/Input/NotReportedIcon.tsx:49]: Well I didnt like the idea of importing smth from publisher to shared components, but I obviously messed up cause I was like partly blind and did not notice specific text. Check this one out [780d463](https://github.com/Recidiviz/justice-counts/pull/650/commits/780d4636dd6bb3b9d7fb6aee497616a7e2ed9ba4)

[the remark this answers is not in the document]
Comment (mxosman) [reply common/components/Input/NotReportedIcon.tsx:49]: I hear you! Haha - no worries at all. I LOVE that - looks great!

**Review by mxosman — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**